### ISSUE TITLE
How Muon's Spectral Design Benefits Generalization: A Study on Imbalanced Data

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#455]
## Info

- How Muon's Spectral Design Benefits Generalization: A Study on Imbalanced Data
  - Bhavya Vasudeva, Puneesh Deora, Yize Zhao, Vatsal Sharan, Christos Thrampoulidis
  - University of Southern California, University of British Columbia
  - https://arxiv.org/abs/2510.22980

### 要約
本論文は、スペクトル正規化を基盤とする最適化手法「Muon」や「Shampoo」が、なぜ一般化性能で従来のSGDやAdamを上回るのかを体系的に分析した研究である。  
著者らは、スペクトル版勾配降下（SpecGD）を理論モデルとして導入し、特にデータ不均衡下において、全ての主成分を等速度で学習する特性が優れた一般化をもたらすことを数理的に示した。  
理論解析と多様な実験（Colored-MNIST, CIFAR, MultiNLI, CelebA, TinyStoriesなど）を通じ、Muonが少数派クラスやスプリアス関連を持つデータに対してバランスの取れた学習を実現することを明らかにした。

---

#### 1. 研究背景
深層学習の分野では、**スペクトル情報を考慮した行列単位の最適化手法（Muon, Shampoo）**が急速に普及している。しかし、それらがどのような条件下で従来のSGDやAdamよりも**良い一般化性能**を示すかは未解明であった。  
従来研究ではAdamの暗黙的バイアス（ℓ∞マージン最大化など）は解明されているが、スペクトル正規化型手法の一般化理論はほとんど存在しなかった

#### 2. 手法の概要：Spectral Gradient Descent（SpecGD）
著者らはMuonの基礎形を抽象化し、**勾配の特異値分解（SVD）を用いて更新方向を U V⊤ とするSpecGD**を導入。  
SpecGDはAdamにおけるSignGDと同様に、**スペクトルノルムに基づく正規化勾配降下**の正準形として機能する
---

### 💡理論的貢献と新規性

1. **不均衡データを用いた解析枠組みの提示**  
   Gaussian混合モデルを用い、クラス不均衡データに対してSpecGDと通常のGDを解析。  
   結果として、GDが支配的成分（多数派クラス）を優先的に学習する一方、SpecGDは**全ての主成分を等速度で学習する**ことを厳密に定式化した:

2. **Theorem 1–2（主要定理）**  
   早期学習段階で、SpecGDはGDおよび正規化GD（NGD）よりも少数派クラスおよびクラスバランス損失で有意に優れることを解析的に証明。  
   損失差が時間に対して線形に成長することを示した（例：ΔL_bal ≥ μt ⁄ 2）

3. **深層線形モデルへの拡張（Proposition 2）**  
   ネットワークの深さ L を増やすことで、各スペクトル成分の飽和点（学習完了時刻）の差が減少し、**深さがバランス学習を加速する**ことを明らかにした（図5–6参照）

---

### 🧠実験的検証

- **Colored-MNIST**（図1・14）  
  数字と色が99%の相関をもつ設定で、Muonは少数派（色の反転グループ）の精度を早期から大幅に改善。  
  図1の右側の画像再構成では、色が優勢な主成分を占め、数字形状情報が下位成分に存在することを可視化し、**Muonがこれらを均等に学習している**ことを示す。

<img width="1045" height="454" alt="Image" src="https://github.com/user-attachments/assets/4bd2818a-6f87-40b1-8048-5560b43c29bd" />

- **CIFAR-10/100 クラス不均衡（図3・17–20）**  
  ResNetモデルにおいて、Muonは少数派クラスの精度を安定的に向上させ、初期学習でのバランス精度が最も高い。

<img width="517" height="567" alt="Image" src="https://github.com/user-attachments/assets/04a82428-7c03-4a0a-a0ba-6d5e2f6a84d1" />

- **グループ不均衡タスク（Dominoes, MultiNLI, CelebA）**  
  MuonおよびShampooはSGDを大きく上回り、特に**スプリアス特徴に依存しない表現学習**を促進（図7–8）。

<img width="791" height="456" alt="Image" src="https://github.com/user-attachments/assets/2340f222-e1b1-4fc7-b6ad-26acf7d33287" />

- **自然言語生成（TinyStories）**  
  頻出トークン（多数派）と希少トークン（少数派）に分けて解析した結果、Muonは両者の学習速度を均等化し、SGDやAdamよりも希少語に対するTop-k精度が高い（図9・25）。

- **属性分類（図26）**  
  「植物・動物」など階層的カテゴリー学習では、GDが粗いクラスのみを早く学習するのに対し、Muonは全階層を同時に学習する傾向を示す。

---

### 🧾検証方法と貢献の実証
- 理論解析では**閉形式の学習軌道（training trajectory）**を導出し、SpecGDとGDの違いを定量化。  
- 実験では**理論的な予測と一致する動的特性**（例えば学習初期の少数派損失ギャップ）を多数のデータセットで確認。  
- また、**Weighted Cross-Entropy (wCE)** を用いた比較により、Muonが暗黙的に再重み付け効果を持つことも示された（図27）。

---

### 🖼️重要なFigureの説明
- **Figure 1:** Colored-MNISTにおけるテスト精度とスペクトル解析。Muonが少数派群で高精度を維持し、スペクトル分解では色成分が支配的であることを可視化。  

<img width="814" height="355" alt="Image" src="https://github.com/user-attachments/assets/9d514db3-7808-4f79-9a41-f2dcf34de01d" />

- **Figure 2:** SpecGDとGDの学習ダイナミクス比較。GDは支配成分を優先、SpecGDは全成分を同速度で学習。  

<img width="346" height="611" alt="Image" src="https://github.com/user-attachments/assets/9259fef5-7cec-4f60-8c06-a54b5eaa187a" />

- **Figure 5–6:** 深さによる影響を可視化。モデルが深くなると成分間の飽和ギャップが縮小。  

<img width="810" height="349" alt="Image" src="https://github.com/user-attachments/assets/5342e7d5-c7dd-4280-8359-bec496946ea3" />

- **Figure 7–9:** 実タスクにおける一般化性能（Dominoes, MultiNLI, CelebA, TinyStories）。Muonの初期一般化優位を実証。

<img width="825" height="321" alt="Image" src="https://github.com/user-attachments/assets/371834a2-1e8b-4466-be34-ad16d8b1a82f" />

---

### 🧭結論と展望
Muonなどのスペクトル最適化は、**データの主成分を均等に学習することで、バランスの取れた一般化**を実現することが理論的・実験的に示された。  
今後の課題は、有限サンプル設定や交差エントロピー損失への拡張、言語モデルなどの系列データにおけるスペクトル構造の同定である

---
